lang/cs/Org.Apache.REEF.Network/Group/Task/Impl/GroupCommClient.cs

```diff
@@ -71,18 +73,21 @@ public sealed class GroupCommClient : IGroupCommClient
             }
 
             networkService.Register(new StringIdentifier(taskId));
+        }
 
+        public void WaitingForRegistration(CancellationTokenSource cancellationSource = null)
+        {
             try
             {
                 foreach (var group in _commGroups.Values)
                 {
-                    group.WaitingForRegistration();
+                    group.WaitingForRegistration(cancellationSource);
                 }
             }
-            catch (ReefRuntimeException e)
+            catch (SystemException e)
```

Comment:
Yes, there is no catch for ReefRuntimeException in the code any more. 

We originally made both ClassHierarchyException and BindException inherit from SystemException, just to make them at system level. The usage seems always catch those two exceptions explicitly. 
In Java REEF, they inherit from RuntimeException.